### ISSUE TITLE
TEST-50-DISSECT: The Return of the Fail 2 (aka the sequel nobody wanted)

### ISSUE THREAD

**Issue by mrc0mmand** [systemd/systemd#24450]
### systemd version the issue has been seen with

latest main

### Used distribution

Arch Linux

### Linux kernel version used

5.19.2-arch1-2

### Unexpected behaviour you saw

Oh well, looks like only the second issue from #24390 was resolved, the first one is still present:

```
[   33.170687] testsuite-50.sh[535]: + systemd-dissect --json=short --root-hash e9633272e0fbb222e5fe59e542eb95723972fdba22cfdee1422ce8ebc1f2ce17 /tmp/tmp.d5Wfbk/minimal_0.gpt
[   33.171400] testsuite-50.sh[536]: + grep -q '{"rw":"ro","designator":"root","partition_uuid":"e9633272-e0fb-b222-e5fe-59e542eb9572","partition_label":"Root Partition","fstype":"squashfs","architecture":"x86-64","verity":"yes",'
[   33.171972] testsuite-50.sh[535]: Opened '/tmp/tmp.d5Wfbk/minimal_0.gpt' in O_RDONLY access mode, with O_DIRECT disabled (O_DIRECT was requested but not supported).
[   33.172623] testsuite-50.sh[535]: Failed to enable O_DIRECT for backing file descriptor for loopback device. Continuing without.
[   33.173404] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/power/uevent" does not exist.
[   33.174054] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/holders/uevent" does not exist.
[   33.174770] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/uevent" does not exist.
[   33.175469] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/uevent" does not exist.
[   33.176217] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu11/uevent" does not exist.
[   33.176949] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu9/uevent" does not exist.
[   33.177681] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu7/uevent" does not exist.
[   33.178377] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu5/uevent" does not exist.
[   33.179114] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu16/uevent" does not exist.
[   33.179843] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu3/uevent" does not exist.
[   33.180572] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu14/uevent" does not exist.
[   33.181304] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu1/uevent" does not exist.
[   33.181968] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu12/uevent" does not exist.
[   33.182744] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu10/uevent" does not exist.
[   33.183468] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu8/uevent" does not exist.
[   33.184176] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu6/uevent" does not exist.
[   33.184850] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu17/uevent" does not exist.
[   33.185609] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu4/uevent" does not exist.
[   33.186338] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu15/uevent" does not exist.
[   33.187132] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu2/uevent" does not exist.
[   33.187786] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu13/uevent" does not exist.
[   33.362213] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/mq/0/cpu0/uevent" does not exist.
[   33.362803] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/queue/uevent" does not exist.
[   33.363400] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/integrity/uevent" does not exist.
[   33.364027] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/trace/uevent" does not exist.
[   33.364693] testsuite-50.sh[535]: sd-device: the uevent file "/sys/devices/virtual/block/loop0/slaves/uevent" does not exist.
[   33.365301] testsuite-50.sh[535]: loop0: Skipping non-partition child (devtype=disk).
[   33.365996] testsuite-50.sh[535]: Successfully acquired /dev/loop0, devno=7:0, nr=0, diskseq=13
[   33.366630] testsuite-50.sh[535]: Kernel was quicker than us in adding partition 1.
[   33.367276] testsuite-50.sh[535]: Kernel was quicker than us in adding partition 2.
[   33.367920] testsuite-50.sh[535]: Kernel was quicker than us in adding partition 3.
[   33.368540] testsuite-50.sh[535]: Probed fstype 'squashfs' on partition /dev/loop0p1.
[   33.369236] testsuite-50.sh[535]: Successfully forked off '(sd-dissect)' as PID 537.
[   33.370076] testsuite-50.sh[537]: Mounting /dev/loop0p1 (squashfs) on /tmp/dissect-jvjHhY (MS_RDONLY|MS_NODEV "")...
[   33.370519] testsuite-50.sh[537]: Failed to mount /dev/loop0p1 (type squashfs) on /tmp/dissect-jvjHhY (MS_RDONLY|MS_NODEV ""): Block device required
[   33.371166] testsuite-50.sh[537]: Failed to mount dissected image: Block device required
[   33.371849] testsuite-50.sh[535]: Failed to read /etc/hostname of image: No such file or directory
[   33.372489] testsuite-50.sh[535]: /etc/machine-id file of image is empty.
[   33.373250] testsuite-50.sh[535]: Failed to read has-init-system boolean: Input/output error
[   33.373772] testsuite-50.sh[535]: (sd-dissect) failed with exit status 1.
[   33.375051] testsuite-50.sh[535]: Failed to acquire image metadata: Block device required
[   36.265257] testsuite-50.sh[535]: Failed to remove device /dev/loop0: Device or resource busy
[   36.265475] testsuite-50.sh[535]: BLKPG_DEL_PARTITION failed, ignoring: No such device or address
[   36.265520] testsuite-50.sh[535]: BLKPG_DEL_PARTITION failed, ignoring: No such device or address
[   36.266231] testsuite-50.sh[386]: + cleanup
[   36.266674] testsuite-50.sh[386]: + '[' -z /tmp/tmp.d5Wfbk ']'
[   36.267270] testsuite-50.sh[386]: + rm -rf /tmp/tmp.d5Wfbk
```

Journal: [system.journal.tar.gz](https://github.com/systemd/systemd/files/9428252/system.journal.tar.gz)

**Comment by bluca** [comment]
_sigh_

block device required? wut?

**Comment by yuwata** [comment]
@mrc0mmand I hope this is fixes by #24530. Could you test with the current HEAD?

**Comment by mrc0mmand** [comment]
> @mrc0mmand I hope this is fixes by #24530. Could you test with the current HEAD?

After another 100 iterations with the current HEAD I can confirm it appears to be fixed, thanks!
